### PR TITLE
MULE-18255: OCSClient must support configuring the api version of the service by system property

### PR DESCRIPTION
**Pull request by ndinu**

### CODE REVIEW

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java [hunk 1]
```diff
@@ -47,6 +47,17 @@
...
   OAuthPlatformManagedDancerBuilder platformUrl(String platformUrl);
 
+  /**
+   * Sets the platform API version to a specific one.
+   *
+   * @param apiVersion the platform API version
+   * @return {@code this} builder
+   * @since 4.3.1 4.4.0
+   */
+  default OAuthPlatformManagedDancerBuilder apiVersion(String apiVersion) {
```
Comment (marianogonzalez) [inline modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java:57]: this method should be introduced in 4.4.0 only. 4.3.x should deal with sys prop only

**Comment by marianogonzalez** [comment]
Issue key shouold be MULE

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConfig.java [hunk 2]
```diff
@@ -62,16 +64,26 @@ public static PlatformManagedOAuthConfig from(String ownerConfigName,
...
                                           delegateGrantType);
   }
 
+  private static String getProperty(ConfigurationProperties configurationProperties, String key, boolean isRequired) {
+    if (isRequired) {
+      return configurationProperties.resolveStringProperty(key)
+          .orElseThrow(() -> new IllegalArgumentException(format("OCS property '%s' has not been set", key)));
+    } else {
+      return configurationProperties.resolveStringProperty(key)
+          .orElse(null);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConfig.java:81]: default value should be v1

Comment (ndinu) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/oauth/ocs/PlatformManagedOAuthConfig.java:81]: discussed privately

File: modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java (hunk 1 above)
Comment (ndinu) [reply modules/oauth-api/src/main/java/org/mule/runtime/oauth/api/builder/OAuthPlatformManagedDancerBuilder.java:57]: Done

**Pull request merged**